2005-03-19  Yoshinori K. Okuji  <okuji@enbug.org>

	* stage2/stage2.c (cmain): Initialize DEFAULT_FILE to an empty
	string. Reported by NATORI Shin <natori@adm.s.u-tokyo.ac.jp>.

http://bugs.gentoo.org/83287

===================================================================
RCS file: /cvsroot/grub/cvsroot/grub/grub/stage2/stage2.c,v
retrieving revision 1.45
retrieving revision 1.46
diff -u -r1.45 -r1.46
--- grub/grub/stage2/stage2.c	2004/07/24 18:58:02	1.45
+++ grub/grub/stage2/stage2.c	2005/03/19 18:02:52	1.46
@@ -872,6 +872,7 @@
 	  
 	  /* Get a saved default entry if possible.  */
 	  saved_entryno = 0;
+	  *default_file = 0;
 	  grub_strncat (default_file, config_file, DEFAULT_FILE_BUFLEN);
 	  for (i = grub_strlen(default_file); i >= 0; i--)
 	    if (default_file[i] == '/')
